### PR TITLE
Provide machine name as a machine's environment variable

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
Provide machine's name in environment variable `CHE_MACHINE_NAME` of machine.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/9653

### CODE REVIEW

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/environment/InternalEnvironmentFactory.java [hunk 1]
```diff
@@ -43,6 +43,7 @@
  * @author Sergii Leshchenko
  */
 public abstract class InternalEnvironmentFactory<T extends InternalEnvironment> {
+  public static final String CHE_MACHINE_NAME_ENV_VAR = "CHE_MACHINE_NAME";
```
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/environment/InternalEnvironmentFactory.java]: Please move it to Constants class

File: wsmaster/che-core-api-workspace-shared/src/main/java/org/eclipse/che/api/workspace/shared/Constants.java [hunk 2]
```diff
@@ -41,6 +41,7 @@
 
   public static final String CHE_WORKSPACE_AUTO_START = "che.workspace.auto_start";
 
+  public static final String CHE_MACHINE_NAME_ENV_VAR = "CHE_MACHINE_NAME";
```
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace-shared/src/main/java/org/eclipse/che/api/workspace/shared/Constants.java]: Please add java doc

**Review by sleshchenko — APPROVED** [review]
LGTM

**Comment by mkuznyetsov** [comment]
ci-test

**Pull request merged**